tests/server.rs

```diff
@@ -1590,31 +1586,40 @@ async fn header_read_timeout_slow_writes_multiple_requests() {
         ",
         )
         .expect("write 4");
-
         thread::sleep(Duration::from_secs(6));
-
+        // idle would timeout in 2s, switched to header read with timeout in 5s
         tcp.write_all(
             b"\
             GET / HTTP/1.1\r\n\
             Something: 1\r\n\
-            \r\n\
         ",
         )
         .expect("write 5");
-        thread::sleep(Duration::from_secs(6));
+        thread::sleep(Duration::from_secs(3));
+        // idle would have timed out, header read times out in 2s
         tcp.write_all(
             b"\
-            Works: 0\r\n\
+            Is: 0\r\n\
+        ",
+        )
+        .expect("write 6");
+        thread::sleep(Duration::from_secs(3));
+        // header read timed out 1s ago
+        tcp.write_all(
+            b"\
+            Wrong: 0\r\n\
         ",
```

Comment:
The header read timeout test was broken (failed due to invalid header after finished request, not timeout). This may explain how the broken read timeout went unnoticed. I have fixed it, and now it doubles as a test in which the idle timeout would trigger if not for the header read timeout.